designs/credentialManagement/credentialManagement.md

```diff
@@ -0,0 +1,139 @@
+project for setting defaults if none are found. At this point the `CredentialManager` should be started, if it is not already.
+
+    Note: Due to being at the project level, another project could have started the `CredentialManager` since it is scoped at the application level.
+* `ValidatingConnection` - Represents that we are performing a [deep credential check](#deep-credential-validation) on the requested _Active Connection Settings_
+* `ValidConnection` - Represents that the deep check passed and the user is told the credentials can be used
+* `IncompleteConfiguration` - Represents the either a region or `CredentialIdentifier` is not selected so we lack the required data to talk to AWS.
+* `InvalidConnection` - Represents that the deep check failed and the user should be told the why and possible remediate actions.
+* `RequiresUserAction` - Represents that we have just left `InitializingToolkit` but the _Active Connection Settings_ will require user action in order for the 
+deeper credential check to proceed such as an MFA prompt. This state is present in order to improve the UX by not forcing the user to perform an action on 
+project opening.
+
+### Picking Defaults
+In order to provide a good "out of the box" experience, the toolkit will attempt to use sane defaults when a project is opened with out any previous settings.
+
+#### Credential Profile
+The Toolkit will attempt to load the `default` profile located in the Shared Credentials files.
+
+An example `~/.aws/config`:
+```ini
+[default]
+credential_process = /usr/bin/myCredentialProcess
+```
+
+#### Region
+The Toolkit attempts to determine a default region based on a heuristic, the region ID must exist in the `endpoints.json` metadata to be considered valid. 
+If a region ID resolved by one step in the heuristic does not exist in the metadata, the Toolkit will continue down the list until a valid region is found.
+
+1. **Last selected (by Project)** - if the _Project_ has previously been opened with the Toolkit - the last region selected when the toolkit closed will be 
+preserved.
+1. **Environment variable / system property** - uses the AWS Java SDK [SystemSettingsRegionProvider] Region Provider to determine region based on the 
+`AWS_REGION` environment variable or `aws.region` system property.
+1. **Default Profile** - uses the AWS Java SDK [AwsProfileRegionProvider] Region Provider to interrogate the `default` profile from the Shared Credentials 
+files, using `region` if found in the profile. 
+    An example `~/.aws/config`:
+    ```ini
+    [default]
+    region = us-west-2
+    ```
+
+1. **us-east-1** - looks for `us-east-1` in resolved metadata.
+1. **First region in metadata** - if all else fails look for the first region that exists in the `endpoints.json` file.
+
+If all of the above fails, the toolkit will throw an exception due to the region data has not been populated and must be considered a fatal error.
+
+### Combined State Flow
+![StateFlow]
```

Comment:
Can you add an outline to the text on stateflow to make it more readable? Otherwise the graph looks good

Reply:
Make diamonds bigger so it is easier to read